Inheritance wallet - Chapter 3 - Wallet backup

rabbitholiness opened the pull request:
This PR adds a page about how the wallet is backed up and how the material is distributed.

👀 [Preview link](https://deploy-preview-1102--bitcoin-design-site.netlify.app/guide/inheritance-wallet/backup/) 👀

mouxdesign commented:
Adding in a review on this one as well. Similar to the previous review zooming out a bit on the content a higher level overview zoomed out would help the content to be easier to chunk in the readers mind. There seem to be 8 larger steps here and then 2 larger buckets. Will dive in more:

## 8 larger steps
### Step 1: Initiate Backup
App prompts user to download the wallet backup kit

### Step 2: Generate Backup Kit (this is titled "what is a backup kit", if rephrased it could be separated into one district step where the kit is generated)
Wallet creates a ZIP file containing:
a. PDF with wallet descriptor and XPUB information
b. Visual representation of recovery rules
c. Wallet backup files in various formats

### Step 3: Download Backup Kit
User saves the backup kit to their device

### Step 4: Secure Backup Kit
User encrypts the ZIP file with a strong password

### Step 5: Store Primary Copies
Users (e.g., Alice and Bob) store copies in their personal password managers

### Step 6: Distribute Backup Components (possible options)
- Store encrypted backup kit with lawyer
- Store password in bank deposit box
- Place backup materials in home safe (USB drive, printout of will and recovery PDF)
- Store safe PIN in shared family password manager

### Step 7: Set Up Secondary Access
- Provide children (heirs) with information on accessing home safe
- Inform children about the assisted route through lawyer and bank

### Step 8: Maintain Backups
Regularly verify and update backups as needed

## More high level
Then these are further separated into actions that are happening in the interface and things that are happening offline. For the online interactions with the interface these might be good to clarify to the person reading the content.
### UI - interactive 
Steps 1-4

### Offline (mostly)
Steps 5-8

A diagram might help to visualize this for a reader so they can zoom out and understand the content at a higher level and connect how each main section fits into each other.

## Summary
These steps could be shown as one high level diagram and then there are two color sections (one color is showing online activities and the other is showing offline activities).

This would be relevant for a UX designer looking at this page so that they understand what they design and what is educational

rabbitholiness commented:
This is a major update. It contains the following main changes:

- Addressed @mouxdesign feedback regarding structure and readability. Thanks very much for the thorough feedback 🙏🏻.
- More concise breakdown of the individual components of the backup scheme.
- Added peivacy and security considerations.
- Added recovery scenarios
- Created mobile optimized images for the diagrams

Still to do:
- [ ] Optimize images (dimensions, size, etc.)
- [ ] Remove unused images from repo
- [ ] Alt texts

Questions:
The page is quite long. Maybe the first section (what needs to be backed up) could be part of the Bitcoin Backups page and I could only reference it here.

danielnordh commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,254 @@
+#### Wallet configuration
+In addition to the private keys, it is also necessary to back up the wallet configuration as such. This is needed to recover a multi-key wallet, because the wallet application needs to know how to generate bitcoin addresses as well as the rules that define how bitcoin can be spent from these addresses. The wallet configuration, also called the [wallet descriptor]({{ '/guide/glossary/#output-scrip ...
+
+#### Don’t store them together
```
This could be presented with a custom tip box as described in the Formatting guide.

{% include tip/open.html color="red" icon="forbid" label="Don't store private keys and wallet descriptor together" %}

If a third party gains access to the descriptor, they can import the wallet to their own application and monitor all past and future activity. However, they cannot spend any funds, as long as they don’t have access to the necessary number of private keys to sign transactions. This is why it is critical that private keys are stored separately from the wallet descriptor.

{% include tip/close.html %}

danielnordh commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,254 @@
+   modalHeightMobile = 2936
+%}
+
+### Primary Access (Self-Recovery)
```
I would simplify this and just call it Self-recovery (or similar, primary and secondary don't add much).

danielnordh commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,254 @@
+### Primary Access (Self-Recovery)
+The primary access setup makes sure that they don’t need anybody else to retrieve the backup material and recover the wallet. To achieve this, Alice and Bob distribute the material between their home safe and the password manager.
+
+#### Home Safe
```
The illustration above says 'House safe', maybe use Home safe everywhere.
Also, sentence case everything on this page (not Home Safe, etc)

danielnordh commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,254 @@
+
+Bob places the following items in a tamper-evident bag and stores it in their home safe:
+
+- A USB drive that contains the recovery kit (unencrypted).
```
The illustration below mentions use being encrypted, but here it says unencrypted.

danielnordh commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,254 @@
+This way, the family can access the material if needed, but any tampering would be immediately noticeable.
+
+
+#### Password Manager
```
I would place this section before 'Home safe' just to be consistent with the illustration which mentions this first. It also explains why the encryption key is held in the safe.

danielnordh commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,254 @@
+- Bank deposit box details
+
+### Inheritance Planning
+Alice and Bob keep a copy of their Last Will in the home safe and give another to Edward, ensuring smooth inheritance processes.
```
Maybe worth including that the will should mention the bitcoin wallet and refer to the recovery documents.

danielnordh commented:
> Questions: The page is quite long. Maybe the first section (what needs to be backed up) could be part of the Bitcoin Backups page and I could only reference it here.

I think you can cut the paragraphs called 'Private keys' and 'Wallet configuration' as they don't add much more than what is already said above in the bullet list and illustration.

I left a bunch of small comments throughout (for some reason I couldn't directly suggest code changes on this PR).

Style comments:
- Sentence case everything (Assisted inheritance), lots of title case (Assisted Inheritance) 
- Lists should not end with punctuation

GBKS commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,279 @@
+Our application goes a bit further than only providing the users with a wallet descriptor. It generates a recovery kit that contains all necessary information in a variety of formats, so that users can easily restore a wallet in case of need. The recovery kit is a ZIP folder that includes:
+
+- **Wallet backup files** compatible with third-party applications like Sparrow, Electrum or Nunchuk
+- **A recovery PDF** containig: 
```
```suggestion
- **A recovery PDF** containing: 
```

GBKS commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,279 @@
+
+1. **Alice and Bob need to recover the wallet themselves**: They might simply need to restore the wallet because one (or both) of them lost access to their phone or computer. This route also accounts for one of the most most likely scenarios in which one spouse outlives the other 
+1. **A self-sovereign inheritance route for Christina and David**: The children should be able to use their inheritance keys and the backup material without any outside help
+1. **An assisted inheritance route** that provides e a fallback in case the self-sovereign inheritance fails. It is designed to help Christina and David access the funds through regular inheritance proceedings. 
```
```suggestion
1. **An assisted inheritance route** that provides a fallback in case the self-sovereign inheritance fails. It is designed to help Christina and David access the funds through regular inheritance proceedings. 
```

GBKS reviewed (changes requested):
Looks great. Just found two typos, then let's go.

Disclaimer, I did not review image details.

mouxdesign commented on guide/inheritance-wallet/onboarding-cosigners.md:
```diff
@@ -0,0 +1,153 @@
+
+{% include image-gallery.html pages = page.images_bob-export-config %}
+
+The app also shows an option to view the [wallet descriptor](https://bitcoin.design/guide/glossary/#output-script-descriptor), which is a small piece of text that contains all the information that a wallet application needs to recover a multi-key wallet. We will cover this topic in more detail in the following chapter about the wallet backup.
```
Earlier on there is mention that the wallet config file is sometimes referred to as the descriptor. It might be easier to in this paragraph keep to one word as the title contains "configuration".

mouxdesign commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,279 @@
+There are two parts to backing up a multi-key wallet: the private keys as well as the wallet configuration.
+
+- **Private keys**: Secure backups of all private keys used in the wallet.
+- **Wallet descriptor**: A small piece of text containing the wallet configuration.
```
Would suggest to use one term or wallet descriptor or wallet configuration since elsewhere it mentions that these can be used interchangeably.

mouxdesign commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,279 @@
+
+##### Alice and Bob's personal vaults
+
+Alice and Bob each store an encrypted version of the recovery kit in their own personal vaults of the password manager they use. Encrypting the recovery kit protects it from a potential security breach on the side of the password manager. 
```
Personal vault and password manager seem almost the same thing here or would there be a difference?

rabbitholiness replied:
Yes, there is a difference:  The password manager is the overall application, while a vault is a dedicated space inside of this application, where users can store passwords/information. In our case, Alice and Bob each have their own space (vault) that only they have access to.  

In addition, they created a shared family vault with shared access for all family members. So everybody will be able to access the material in the family vault, but not the material in Alice's and Bob's personal vaults.

mouxdesign commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,279 @@
+
+##### Shared family vault
+
+To make sure that Christina and David will be able to retrieve both the recovery kit  and the password to decrypt it, Alice and Bob store the following pieces of information in a shared family vault:
```
```suggestion
To make sure that Christina and David will be able to retrieve both the recovery kit and the password to decrypt it, Alice and Bob store the following pieces of information in a shared family vault:
```

mouxdesign commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,279 @@
+
+### Home safe
+
+Bob places the following items in a tamper-evident bag and stores it in their home safe:
```
Would this be the "vault" referred to above?

rabbitholiness replied:
It's a bit tricky, because the terminology is quite similar for both concepts. But one is physical and the other is digital: 

- the home safe is the physical safe that the Joneses have in their home
-  vaults are digital vaults inside of the password manager application. 

I've tried to visualize this in the "backup material distribution" section and diagram.

mouxdesign commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,279 @@
+
+The setup oulined above addresses some important aspects of wallet backups:
+
+1. **Tamper-evidence**: The tamper-evident bag in the home safe allows Alice and Bob to quickly check if anyone has attempted to access the backup materials
```
Vault?
```suggestion
1. **Tamper-evidence**: The tamper-evident bag in the home safe allows Alice and Bob to quickly check if anyone has attempted to access the backup materials
```

rabbitholiness replied:
Here we are talking about the physical safe in Alice's and Bob's home. Vaults refer to digital vaults in the password manager application.

mouxdesign commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,279 @@
+If anything happens to Alice and Bob, Christina and David can access the funds after six months using:
+
+- The PIN to the safe, stored in the shared family vault of the password manager
+- The backup materials from the home safe
```
```suggestion
- The backup materials from the personal vault
```

Suggestion here is to keep the word vault consistent throughout the copy. Variations in the copy are "tamper evident bag", "Home safe", "backup materials".

rabbitholiness replied:
As mentioned in the previous comments, "safe" and "vault" refer to two different types of vaults. I made some text updates that helps clarify this a bit.

mouxdesign reviewed (commented):
Looks great overall, added some minor copy suggestions for consistency in the copy.

rabbitholiness commented:
Just resolved the conflicts, @GBKS . Should be good to go now.

GBKS commented on guide/inheritance-wallet/backup.md:
```diff
@@ -0,0 +1,252 @@
+
+To achieve this goal, their backup strategy accounts for the following three scenarios: 
+
+1. **Alice and Bob need to recover the wallet themselves**: They might simply need to restore the wallet because one (or both) of them lost access to their phone or computer. This route also accounts for one of the most most likely scenarios in which one spouse outlives the other 
```
```suggestion
1. **Alice and Bob need to recover the wallet themselves**: They might simply need to restore the wallet because one (or both) of them lost access to their phone or computer. This route also accounts for one of the most likely scenarios in which one spouse outlives the other 
```

GBKS reviewed (changes requested):
Looking good. Just some minor stuff now.

Small tweak would also be to update the "next page" link in the "Cosigner onboarding" page to go to this new page. It still takes you to the "Upgradeable wallet" right now.

Please review the images one more time. Sizes in markdown don't match the actual files.  The files sizes themselves are also a bit all over the place, we have recommended sizes listed [here](https://bitcoin.design/guide/contribute/illustration-guidelines/#export-images-at-the-right-sizes).

GBKS reviewed (approved):
ACK. Let's go.
